custom_components/ha_washdata/ml/engine.py

```
@@ -165,6 +187,21 @@ def resolve_regressor(capability: str, store: object | None):
+                from .feature_extraction import PROGRESS_FEATURE_COLUMNS
+                _expected_r = list(PROGRESS_FEATURE_COLUMNS)
+                _stored_r = list(spec.get("feature_columns") or [])
+                if _expected_r and _stored_r and _stored_r != _expected_r:
+                    _LOGGER.warning(
+                        "Promoted regression spec for %r has stale feature schema "
+                        "(%d cols vs current %d); reverting to inert.",
+                        capability, len(_stored_r), len(_expected_r),
+                    )
+                    return (None, None)
+            except Exception:  # noqa: BLE001 - schema check must not break inference
+                pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)